libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountBuilder.kt

```diff
@@ -0,0 +1,529 @@
+        return if (!allowUnset && contentSid == null) this else apply { this.contentSid = contentSid }
+    }
+
+    /**
+     * Sets CSRF token.
+     *
+     * @param csrfToken CSRF token.
+     * @return Instance of this class.
+     */
+    fun csrfToken(csrfToken: String?): UserAccountBuilder {
+        return if (!allowUnset && csrfToken == null) this else apply { this.csrfToken = csrfToken }
+    }
+
+    /**
+     * Sets if the user authenticated with native login.
+     *
+     * @param isNativeLogin if the user authenticated with native login.
+     * @return Instance of this class.
+     */
+    fun nativeLogin(isNativeLogin: Boolean): UserAccountBuilder {
+        this.nativeLogin = isNativeLogin
+        return this
+    }
+
+    /**
+     * Sets language.
+     *
+     * @param language User's language.
+     * @return Instance of this class.
+     */
+    fun language(language: String?): UserAccountBuilder {
+        return if (!allowUnset && language == null) this else apply { this.language = language }
+    }
+
+    /**
+     * Sets locale.
+     *
+     * @param locale User's locale.
+     * @return Instance of this class.
+     */
+    fun locale(locale: String?): UserAccountBuilder {
+        return if (!allowUnset && locale == null) this else apply { this.locale = locale }
+    }
+
+
+    /**
+     * Sets cookie client src
+     *
+     * @param cookieClientSrc cookie client src.
+     * @return Instance of this class.
+     */
+    fun cookieClientSrc(cookieClientSrc: String?): UserAccountBuilder {
+        return if (!allowUnset && cookieClientSrc == null) this else apply { this.cookieClientSrc = cookieClientSrc }
+    }
+
+    /**
+     * Sets cookie sid client
+     *
+     * @param cookieSidClient cookie sid client.
+     * @return Instance of this class.
+     */
+    fun cookieSidClient(cookieSidClient: String?): UserAccountBuilder {
+        return if (!allowUnset && cookieSidClient == null) this else apply { this.cookieSidClient = cookieSidClient }
+    }
+
+    /**
+     * Sets sid cookie name
+     *
+     * @param sidCookieName sid cookie name.
+     * @return Instance of this class.
+     */
+    fun sidCookieName(sidCookieName: String?): UserAccountBuilder {
+        return if (!allowUnset && sidCookieName == null) this else apply { this.sidCookieName = sidCookieName }
+    }
+
+    /**
+     * Sets oauth client id
+     *
+     * @param clientId sid cookie name.
+     * @return Instance of this class.
+     */
+    fun clientId(clientId: String?): UserAccountBuilder {
+        return if (!allowUnset && clientId == null) this else apply { this.clientId = clientId }
+    }
+
+    /**
+     * Sets additional OAuth values.
+     *
+     * @param additionalOauthValues Additional OAuth values.
+     * @return Instance of this class.
+     */
+    fun additionalOauthValues(additionalOauthValues: Map<String, String>?): UserAccountBuilder {
+        if (!allowUnset) {
```

Comment:
The merge logic is important - and there is a new test to make sure it is right.